### PR TITLE
Benchmarks

### PR DESCRIPTION
**Pull request by mwalmsley**
It's important that Zoobot has proper benchmarks so that we can be confident new releases work properly for users. This PR adds those benchmarks.

In the course of setting up the benchmarks, I have made some major changes/improvements:
- pytorch-galaxy-datasets refactored to work for tensorflow, imports adapted
- both tensorflow and pytorch zoobot versions use albumentations for augmentations. Old TF code removed.
- tensorflow version bumped to 2.10 (current latest) while I'm at it
- pytorch version now has logging for per-question loss. Loss func aggregation has new option to support this.
- TensorFlow version has per-question logging also, but awaiting issue with Keras team to enable
- Created minimal_example.py for TensorFlow (thanks, @katgre )
- Support CPU-only PyTorch training
- Refactor TF TrainingConfig to Trainer object, Lightning style, for consistency

### CODE REVIEW

**Comment by mwalmsley** [comment]
This will cause dev CI to fail until I also PR and push to PyPI the new version of pytorch-galaxy-datasets, cc @camallen

**Comment by mwalmsley** [comment]
Now includes finetuning, galaxy-datasets dependency, and QoL improvements - see #59

**Comment by mwalmsley** [comment]
PyTorch benchmarks run successfully - punching merge to dev
